ext/sockets/sockets.c

```diff
@@ -571,9 +571,14 @@ PHP_FUNCTION(socket_select)
 	zend_long		sec, usec = 0;
 	bool		sec_is_null = 0;
 
-	if (zend_parse_parameters(ZEND_NUM_ARGS(), "a!a!a!l!|l", &r_array, &w_array, &e_array, &sec, &sec_is_null, &usec) == FAILURE) {
-		RETURN_THROWS();
-	}
+	ZEND_PARSE_PARAMETERS_START(4, 5)
+		Z_PARAM_ARRAY_EX2(r_array, 1, 1, 0)
```

Comment:
You sure about this change? This now derefs...

Reply:
yes

```php
function socket_select(?array &$read, ?array &$write, ?array &$except, ?int $seconds, int $microseconds = 0): int|false {}
```

Reply:
Right of course